python/lsst/ts/salobj/sal_info.py

```diff
@@ -37,12 +37,15 @@
 
 MAX_RESULT_LEN = 256  # max length for result field of an Ack
 
-# The python logger remembers the level for a given named log
-# so the log level needs to be initialized every time
-INITIAL_LOG_LEVEL = logging.INFO
-
-# default time to wait for historical data (sec)
-# override by setting env var $LSST_DDS_HISTORYSYNC
+# We want DDS logMessage messages for at least INFO level messages
+# so if the current level is less verbose, set it to INFO.
+# Do not change the level if it is already more verbose,
+# because somebody has intentionally increased verbosity
+# (a common thing to do in unit tests).
+MAX_LOG_LEVEL = logging.INFO
+
+# Default time to wait for historical data (sec);
+# override by setting env var $LSST_DDS_HISTORYSYNC.
```

Comment:
It really baffles me that we need a 1 minutes delay to read historical data.

Reply:
That's DDS for you. It takes awhile for a new Domain Participant to "align" itself with the rest of the system. I hope we can reduce this a bit by setting masterPriority (as SAL 4.2 will do) but I doubt it will help enough to reduce this.

Two things to keep in mind:
* This is an upper limit; if late joiner data arrives early then the SAL components will start quickly.
* If we give up too early then we will get stale data masquerading as new data, which is a big problem (especially for commands).

I hope setting masterPriority will lower startup time for scripts, and startup time for components in the full system (once we increase masterPriority for some component that is started early). But even if it does, a long delay is useful for unit tests, where there is no existing master.